### PR TITLE
viz: Remove online dependency for created files and reduced the size of those files

### PR DESCRIPTION
**Pull request by boeddeker**

### CODE REVIEW

File: meeteval/viz/visualize.py [hunk 1]
```diff
@@ -506,10 +570,31 @@ def html(self):
...
+        def load_cdn(name, url):
+            file = cache_dir / name
+            try:
+                return file.read_text()
+            except FileNotFoundError:
+                cache_dir.mkdir(parents=True, exist_ok=True)
+                urllib.request.urlretrieve(url, file)
+                return file.read_text()
+
+        if False and self.js_debug:
```
Comment (thequilo) [inline meeteval/viz/visualize.py:588]: Remove this branch?

File: meeteval/io/seglst.py [hunk 2]
```diff
@@ -175,12 +175,25 @@ def keys(self):
                 *[set(s.keys()) for s in self._outer.segments]
             )
 
+        def all_keys(self):
```
[the remark this answers is not in the document]
Comment (boeddeker) [reply meeteval/io/seglst.py]: After offline discussion: Merged with keys

File: meeteval/io/seglst.py [hunk 3]
```diff
@@ -175,12 +175,25 @@ def keys(self):
...
+
         def __getitem__(self, key: _SegLstSegment_keys):
             """
             Returns the values for `key` of all segments as a list.
             """
             return [s[key] for s in self._outer.segments]
 
+        def get(self, key, default=None):
+            """
+            Returns the values for `key` of all segments as a list.
```
[the remark this answers is not in the document]
Comment (boeddeker) [reply meeteval/io/seglst.py]: done

File: meeteval/viz/visualize.py (hunk 1 above)
Comment (boeddeker) [reply meeteval/viz/visualize.py]: done

File: meeteval/viz/visualize.py [hunk 4]
```diff
@@ -449,6 +509,10 @@ def _ipython_display_(self):
         ref_session_ids = reference.unique('session_id')
         hyp_session_ids = hypothesis.unique('session_id')
 
+        if self.js_debug:
+            print('WARNING: js_debug is not supported in ipynb. Disable it.')
```
[the remark this answers is not in the document]
Comment (boeddeker) [reply meeteval/viz/visualize.py]: done

**Comment by boeddeker** [comment]
For documentation:
dipco S28 changed from 6,0MB to 1,4MB from the restructured data layout

**Review by thequilo — APPROVED** [review]

File: meeteval/io/seglst.py [hunk 5]
```diff
@@ -165,13 +165,20 @@ class T:
         def __init__(self, outer):
             self._outer = outer
 
-        def keys(self):
+        def keys(self, *, all=False):
             """
-            The keys that are common among all segments
+            The keys that are common among all segments (all=False) or all
+            keys (all=False).
```
[the remark this answers is not in the document]
Comment (boeddeker) [reply meeteval/io/seglst.py]: done

File: meeteval/viz/visualize.js [hunk 6]
```diff
@@ -1178,14 +1242,17 @@ class CanvasPlot {
...
 
-            drag(this.plot.element, e => {
-                const delta = this.plot.y.invert(e.movementY) - this.plot.y.invert(0);
-                let [begin, end] = this.plot.y.domain();
-                this._callOnScrollHandlers(begin - delta, end - delta);
+            drag(this.plot.element, (e, delta_y) => {
+                if (delta_y){
+                    const delta = this.plot.y.invert(delta_y) - this.plot.y.invert(0);
+                    let [begin, end] = this.plot.y.domain();
+                    console.log(delta_y, delta, begin - delta, end - delta, e)
```
[the remark this answers is not in the document]
Comment (boeddeker) [reply meeteval/viz/visualize.js]: done

**Pull request merged**